.github/actions/dependencies/action.yml

```diff
@@ -11,23 +13,33 @@ runs:
       run: |
         conan export --version 1.1.10 external/snappy
         conan export --version 4.0.3 external/soci
-    - name: add Ripple Conan remote
-      if: env.CONAN_URL != ''
+    - name: add Conan remotes
+      if: env.CONAN_REMOTE_URL != ''
       shell: bash
       run: |
-        if conan remote list | grep -q "ripple"; then
-            conan remote remove ripple
-            echo "Removed conan remote ripple"
+        echo "Adding Conan remotes for recipes."
+        if conan remote list | grep -q "recipes"; then
+          conan remote update recipes --index 0 --url ${{ env.CONAN_REMOTE_URL }}/recipes
+          echo "Updated Conan remote 'recipes' to ${{ env.CONAN_REMOTE_URL }}/recipes."
+        else
+          conan remote add --index 0 recipes ${{ env.CONAN_REMOTE_URL }}/recipes
+          echo "Added new conan remote 'recipes' at ${{ env.CONAN_REMOTE_URL }}/recipes."
+        fi
+        echo "Adding Conan remotes for packages."
+        if conan remote list | grep -q "${{ inputs.remote_name }}"; then
+          conan remote update ${{ inputs.remote_name }} --index 1 --url ${{ env.CONAN_REMOTE_URL }}/${{ inputs.remote_name }}
+          echo "Updated Conan remote '${{ inputs.remote_name }}' to ${{ env.CONAN_REMOTE_URL }}/${{ inputs.remote_name }}."
+        else
+          conan remote add --index 1 ${{ inputs.remote_name }} ${{ env.CONAN_REMOTE_URL }}/${{ inputs.remote_name }}
+          echo "Added new conan remote '${{ inputs.remote_name }}' at ${{ env.CONAN_REMOTE_URL }}/${{ inputs.remote_name }}."
```

Comment:
I don't know enough about `conan remote` but how do you know that `--index 1` here is the right one to use ?

Reply:
Doesn't this mean that `conan remote add --index 1` should be preceded by `conan remote remove --index 1` or similar, if that slot is already taken by a different remote ?